src/main/java/it/pagopa/transactions/commands/handlers/TransactionAddUserReceiptHandler.java

```diff
@@ -19,17 +21,20 @@
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 import reactor.core.publisher.Mono;
 
+import java.time.Duration;
 import java.time.OffsetDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Locale;
 
 @Component
 @Slf4j
 public class TransactionAddUserReceiptHandler
-        implements CommandHandler<TransactionAddUserReceiptCommand, Mono<TransactionUserReceiptAddedEvent>> {
+        implements
+        CommandHandler<TransactionAddUserReceiptCommand, Mono<Either<Mono<TransactionUserReceiptAddErrorEvent>, Mono<TransactionUserReceiptAddedEvent>>>> {
```

Comment:
Is it possible to simplify this to Mono<Either<TransactionUserReceiptAddErrorEvent,TransactionUserReceiptAddedEvent>>> let's say, without the deepest Mono?

Reply:
Good catch. Refactored with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/179/commits/cd54021047de4c2dc51fcc2a722b2cc3700fc14e